Dev

orthagonal opened the pull request.

orthagonal committed:
first checkin

orthagonal committed:
fixed

orthagonal committed:
basic version is working

probably needs some more error testing

orthagonal committed:
replace async.each with async.map for clarity

jgallen23 commented on index.js:
```diff
@@ -0,0 +1,23 @@
+    async.map(listToProcess, (currentItem, currentItemDone) => {
+      async.auto(createAutoSpec(currentItem), (autoErr, anAutoResult) => {
+        if (autoErr) {
+          throw autoErr;
```
this should be `reduceResults(autoErr)`

jgallen23 commented on index.js:
```diff
@@ -0,0 +1,23 @@
+module.exports = (createList, createAutoSpec, handleAutoResult, reduceResults) => {
+  createList((err, listToProcess) => {
+    if (err) {
+      throw err;
```
this should be `reduceResults(err)`

jgallen23 commented on index.js:
```diff
@@ -0,0 +1,23 @@
+// the third param is a callback function that combines the original listed item with the
+// result of applying async.auto to that item
+// fourth param takes in all results and processes all of them:
+module.exports = (createList, createAutoSpec, handleAutoResult, reduceResults) => {
```
instead of `reduceResults`, lets call it `done`

jgallen23 replied:
instead of handleAutoResult, lets call it `mapAutoResult`

orthagonal committed:
returns error back up chain

returns error back up the callback chain instead of calling 'throw'
changed some function names for legibility

orthagonal committed:
added travis support
